app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -1,43 +1,55 @@
 # frozen_string_literal: true
+
+# call with
+#  use_work_ids an array is just for testing a few ids
+# HykuAddons::ReindexModelJob.perform_now("GenericWork", "repo.hyku.docker", limit: 1, options: { cname_doi_mint: ["repo.hyku.docker"], use_work_ids: ["15b3c16b-8183-4d2a-86ef-d144176fa0c8"]} )
+#
+# HykuAddons::ReindexAvailableWorksJob.perform_later(["repo.hyku.docker"], options: { cname_doi_mint: ["repo.hyku.docker"]} )
+
 module HykuAddons
   class ReindexModelJob < ApplicationJob
-    queue_as :reindex
-
     rescue_from Hyrax::DOI::DataCiteClient::Error, Ldp::Gone, Ldp::HttpError, RSolr::Error::Http, RSolr::Error::ConnectionRefused do |exception|
       Rails.logger.debug exception.inspect
       retry_job(wait: 5.minutes)
     end
 
-    def perform(klass, cname, limit: 25, page: 1, cname_doi_mint: [])
+    def perform(klass, cname, limit: 25, page: 1, options: {})
+      options ||= options || { cname_doi_mint: [], use_work_ids: [] }
       # for whatever in private methods reason without assigning it to instamce variable it throws undefined local variable
-      @cname_doi_mint = cname_doi_mint
+      @cname_doi_mint = options[:cname_doi_mint]
+      @use_work_ids = options[:use_work_ids]
       @cname = cname
+      @limit = limit
+      @page = page
+      @klass = klass
       AccountElevator.switch!(cname)
 
-      Rails.logger.debug "=== Starting to reindex #{klass} in #{cname} ==="
-
-      offset = (page - 1) * limit
-
-      # When the offset becomes too large, no records would be found
-      works =  klass.constantize.where("title_tesim:*").limit(limit).offset(offset)
-
-      # with calling to_a it always returns true, even when no records found
-      if works.to_a.any?
-        reindex_works(works)
+      offset = (page - 1) * @limit
+      work_class = klass.constantize
+      fetch_work_using_ids(work_class) if @use_work_ids.present?
```

Comment:
I realise Rubocop gave you a warning about method length, but it could be resolved by removing the `offset` variable: 
```
work_class = klass.constantize
if @use_work_ids.present?
      fetch_work_using_ids(work_class) 
else
     # The offset variable is documented by the method signature
      reindex_and_mint_work(work_class, (page - 1) * @limit) 
end
```

It's better to add  `#rubocop:disable Metrics/MethodLength` and have a normal condition than two end of line conditions for `if` and `unless`.

Reply:
@prdanelli one of the point of the Rubocops is to help to provide pointers to things to refactor. So i don't really think we should always defer to adding **#rubocop:disable Metrics/MethodLength** rather than actually refactor the code. Otherwise what is point of the rubocop.

I am not convinced of the benefits of this change vs what is there?